R/interop-car.R

```diff
@@ -33,42 +33,54 @@ h_get_contrast <- function(object, effect, type = c("II", "III", "2", "3"), tol
   type <- match.arg(type)
   mx <- component(object, "x_matrix")
   asg <- attr(mx, "assign")
+  has_intercept <- !0 %in% asg
   formula <- object$formula_parts$model_formula
   tms <- terms(formula)
   fcts <- attr(tms, "factors")[-1L, , drop = FALSE] # Discard the response.
   ods <- attr(tms, "order")
   assert_subset(effect, colnames(fcts))
   idx <- which(effect == colnames(fcts))
   cols <- which(asg == idx)
-  data <- model.frame(object)
-  var_numeric <- vapply(data, is.numeric, FUN.VALUE = TRUE)
-  coef_rows <- length(cols)
+  xlev <- component(object, "xlev")
+  is_intercet <- has_intercept && identical(names(xlev)[1], effect)
```

Comment:
Better `has_intercept`, can also overwrite previous variable or define correctly above?

Reply:
this `is_intercept` means that the current effect is taking the role of intercept. typo here. but should be different from "has_intercept" (which indicates if the model has intercept).

e.g. a formula like `fev1 ~ armcd*avisit - 1`, we will not have the intercept, and only armcd will be expanded to armcdtrt and armcdpbo. avisit will not be expanded

Reply:
so previous `has_intercept` is checking the model while `is_intercept` is checking the current variable

Reply:
this part is not finished because there are complications of model formula, like `a ~ b*c -1`, `a ~ b:c - 1`, `a ~ b*c - b - 1`, and they might have effect on the model matrix/terms etc. still need some time to fix them all

Reply:
ok so I wait for your updates